src/dir_set.mli

```diff
@@ -69,5 +75,12 @@ val diff : 'w t -> 'w t -> 'w t
 val negate : 'w t -> 'w t
 
 val to_sexp : 'w t -> Sexp.t
+val to_dyn : 'w t -> Dyn.t
+
+val forget_root : 'w t -> Path.Unspecified.w t
+
+val toplevel_subdirs : 'w t -> [`Infinite  | `Finite of String.Set.t]
```

Comment:
Small nitpick: I'm personally not a huge fan of polymorphic variants. I remember seeing an example where an unreachable match case was not detected because of using polymorphic variants. Normal variants feel more sturdy

Reply:
Changed to a normal variant.